addons/account_peppol/models/res_config_settings.py

```diff
@@ -1,25 +1,44 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from base64 import b64encode
+from odoo import _, api, fields, models, modules, tools
+from odoo.exceptions import UserError, ValidationError
 
-from odoo import api, fields, models, _
 from odoo.addons.account_edi_proxy_client.models.account_edi_proxy_user import AccountEdiProxyError
-from odoo.exceptions import UserError, ValidationError
+from odoo.addons.account_edi_ubl_cii.models.account_edi_common import EAS_MAPPING
+from .peppol_constraints import PEPPOL_ENDPOINT_WARNING, _check_peppol_endpoint_number
+
+# at the moment, only European countries are accepted
+ALLOWED_COUNTRIES = set(EAS_MAPPING.keys()) - {'AU', 'SG', 'NZ'}
 
 
 class ResConfigSettings(models.TransientModel):
     _inherit = 'res.config.settings'
 
+    account_peppol_edi_user = fields.Many2one(
+        comodel_name='account_edi_proxy_client.user',
+        string='EDI user',
+        compute='_compute_account_peppol_edi_user',
+    )
+    account_peppol_contact_email = fields.Char(related='company_id.account_peppol_contact_email', readonly=False)
     account_peppol_eas = fields.Selection(related='company_id.peppol_eas', readonly=False)
+    account_peppol_edi_identification = fields.Char(related='account_peppol_edi_user.edi_identification')
     account_peppol_endpoint = fields.Char(related='company_id.peppol_endpoint', readonly=False)
-    account_peppol_proxy_state = fields.Selection(related='company_id.account_peppol_proxy_state')
+    account_peppol_endpoint_warning = fields.Char(
+        string="Warning",
+        compute="_compute_account_peppol_endpoint_warning",
+    )
+    account_peppol_migration_key = fields.Char(related='account_peppol_edi_user.peppol_migration_key', readonly=False)
+    account_peppol_phone_number = fields.Char(related='company_id.account_peppol_phone_number', readonly=False)
```

Comment:
Do we really need to store that? Can't we just fill the value, click the button and that's it without storing anything on the company ?

Reply:
we need to show later what phone number we used to send the verification code to, I think this is the easiest way to do this?